### PR TITLE
Merge database tests branch

### PR DESCRIPTION
**Pull request by Manuri**
## Purpose
$Subject

### CODE REVIEW

File: .testgrid-database.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# TestGrid Test Configuration Parameters
+version: '0.9'
+emailToList: "manurip@wso2.com"
```
Comment (daneshk) [inline .testgrid-database.yaml]: shall we add builder@ mail as well here

File: database/mssql/src/test/java/org/ballerinalang/scenario/test/database/mssql/BatchUpdateTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Properties;
+
+public class BatchUpdateTest extends ScenarioTestBase {
```
Comment (daneshk) [inline database/mssql/src/test/java/org/ballerinalang/scenario/test/database/mssql/BatchUpdateTest.java]: Shall we add doc comment

File: database/mssql/src/test/resources/bal-src/batch-update-test.bal [hunk 3]
```diff
@@ -0,0 +1,142 @@
+import ballerina/config;
```
Comment (daneshk) [inline database/mssql/src/test/resources/bal-src/batch-update-test.bal]: license header missing

File: database/mssql/src/test/resources/sql-src/cleanup-call-test.sql [hunk 4]
```diff
@@ -0,0 +1,36 @@
+DROP PROCEDURE CALL_TEST_IN_INTEGER_TYPES;
```
Comment (daneshk) [inline database/mssql/src/test/resources/sql-src/cleanup-call-test.sql]: license header missing

File: database/mysql/pom.xml [hunk 5]
```diff
@@ -0,0 +1,948 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (daneshk) [inline database/mysql/pom.xml]: license header missing

File: database/mssql/src/test/resources/sql-src/dml-select-test.sql [hunk 6]
```diff
@@ -0,0 +1,25 @@
+INSERT INTO SELECT_UPDATE_TEST_INTEGER_TYPES VALUES(1, 32767, 9223372036854775807, 255, 1, 2147483647);
```
Comment (daneshk) [inline database/mssql/src/test/resources/sql-src/dml-select-test.sql]: license header missing

File: database/mysql/src/test/resources/bal-src/batch-update-test.bal [hunk 7]
```diff
@@ -0,0 +1,157 @@
+import ballerina/config;
```
Comment (daneshk) [inline database/mysql/src/test/resources/bal-src/batch-update-test.bal]: License header missing

File: database/oracle/src/test/resources/bal-src/batch-update-test.bal [hunk 8]
```diff
@@ -0,0 +1,146 @@
+import ballerina/config;
```
Comment (daneshk) [inline database/oracle/src/test/resources/bal-src/batch-update-test.bal]: License header missing

File: database/oracle/src/test/resources/bal-src/select-test.bal [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+    string? ncharVal;
+    string? varcharVal;
+    string? nvarcharVal;
+};
+
+type DateTimeTypeStr record {
+    string? dateStr;
+    string? timestampStr;
+   // string? timestampzStr;
+   // string? timestampzLocalStr;
```
Comment (daneshk) [inline database/oracle/src/test/resources/bal-src/select-test.bal]: shall we remove these commented codes

File: database/postgresql/pom.xml [hunk 10]
```diff
@@ -0,0 +1,714 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (daneshk) [inline database/postgresql/pom.xml]: License header missing

File: database/postgresql/src/test/resources/bal-src/batch-update-test.bal [hunk 11]
```diff
@@ -0,0 +1,146 @@
+import ballerina/config;
```
Comment (daneshk) [inline database/postgresql/src/test/resources/bal-src/batch-update-test.bal]: License header missing

File: database/postgresql/src/test/resources/bal-src/call-test.bal [hunk 12]
```diff
@@ -0,0 +1,162 @@
+import ballerina/config;
```
Comment (daneshk) [inline database/postgresql/src/test/resources/bal-src/call-test.bal]: License header missing

File: database/postgresql/src/test/resources/bal-src/select-test.bal [hunk 13]
```diff
@@ -0,0 +1,191 @@
+import ballerina/config;
```
Comment (daneshk) [inline database/postgresql/src/test/resources/bal-src/select-test.bal]: License header missing

File: database/postgresql/src/test/resources/sql-src/cleanup-select-test.sql [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+/
+DROP TABLE IF EXISTS SELECT_UPDATE_TEST_STRING_TYPES;
+/
+DROP TABLE IF EXISTS SELECT_UPDATE_TEST_COMPLEX_TYPES;
+/
+DROP TABLE IF EXISTS SELECT_UPDATE_TEST_DATETIME_TYPES;
+/
+DROP TABLE IF EXISTS UPDATE_TEST_GENERATED_KEYS_NO_KEY;
+/
+DROP TABLE IF EXISTS UPDATE_TEST_GENERATED_KEYS;
```
Comment (daneshk) [inline database/postgresql/src/test/resources/sql-src/cleanup-select-test.sql]:
```suggestion
DROP TABLE IF EXISTS UPDATE_TEST_GENERATED_KEYS;

```

File: database/postgresql/src/test/resources/sql-src/db-init.sql [hunk 15]
```diff
@@ -0,0 +1,3 @@
+CREATE DATABASE testdb;
+/
+\c testdb;
```
Comment (daneshk) [inline database/postgresql/src/test/resources/sql-src/db-init.sql]:
```suggestion
\c testdb;

```

File: database/postgresql/src/test/resources/sql-src/ddl-call-test.sql [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+' LANGUAGE plpgsql
+ /
+CREATE FUNCTION CALL_TEST_INOUT_FLOAT_TYPES(IN p_id_in INTEGER, IN p_id_out INTEGER, INOUT p_realVal REAL, INOUT p_doubleVal DOUBLE PRECISION) AS '
+BEGIN
+    INSERT INTO SELECT_UPDATE_TEST_FLOAT_TYPES VALUES(p_id_in, p_realVal);
+    SELECT VARCHAR_VAL INTO p_realVal FROM SELECT_UPDATE_TEST_FLOAT_TYPES WHERE ID = p_id_out;
+    SELECT TEXT_VAL INTO p_doubleVal FROM SELECT_UPDATE_TEST_FLOAT_TYPES WHERE ID = p_id_out;
+END
+' LANGUAGE plpgsql
+
```
Comment (daneshk) [inline database/postgresql/src/test/resources/sql-src/ddl-call-test.sql]: shall we remove extra new lines

File: database/postgresql/src/test/resources/sql-src/ddl-test.sql [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+(
+    X INTEGER,
+    Y VARCHAR(20)
+);
+/
+CREATE FUNCTION DDL_TEST_DROPPING_PROC(IN X INTEGER, OUT Y VARCHAR(50), INOUT Z VARCHAR(10)) AS
+'BEGIN
+    SELECT "DDL_TEST_CREATE_PROC called";
+END'  LANGUAGE plpgsql;
+/
```
Comment (daneshk) [inline database/postgresql/src/test/resources/sql-src/ddl-test.sql]:
```suggestion
/

```

File: database/util/pom.xml [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+            <groupId>org.ballerinalang</groupId>
+            <artifactId>ballerina-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+    </dependencies>
+
+
```
Comment (daneshk) [inline database/util/pom.xml]:
```suggestion
```

File: scenarios-commons/src/main/java/org/ballerinalang/scenario/test/common/database/DatabaseUtil.java [hunk 19]
```diff
@@ -0,0 +1,58 @@
+package org.ballerinalang.scenario.test.common.database;
```
Comment (daneshk) [inline scenarios-commons/src/main/java/org/ballerinalang/scenario/test/common/database/DatabaseUtil.java]: License header missing

File: scenarios-commons/src/main/java/org/ballerinalang/scenario/test/common/database/DatabaseUtil.java [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+            String line = buf.readLine();
+            StringBuilder sb = new StringBuilder();
+            while (line != null) {
+                sb.append(line).append(System.lineSeparator());
+                line = buf.readLine();
+            }
+            return sb.toString();
+        }
+    }
+}
```
Comment (daneshk) [inline scenarios-commons/src/main/java/org/ballerinalang/scenario/test/common/database/DatabaseUtil.java]:
```suggestion
}

```

File: database/mssql/src/test/resources/sql-src/cleanup-call-test.sql (hunk 4 above)
Comment (Manuri) [reply database/mssql/src/test/resources/sql-src/cleanup-call-test.sql]: Adding the license header would make the script execution logic break, so this was intentional.

File: .testgrid-database.yaml (hunk 1 above)
Comment (Manuri) [reply .testgrid-database.yaml]: As mentioned offline, waiting for approval for a separate mailing list.

**Pull request merged**